backend/lambdas/process_post/lambda.py

```diff
@@ -0,0 +1,193 @@
+
+import demoji
+
+import boto3
+import langchain_core
+from langchain_aws import ChatBedrock
+
+from output_models.models import ExtractedInformation, TopicMatch
+from prompt_selector import get_information_extraction_prompt_selector, get_topic_match_prompt_selector
+
+from aws_lambda_powertools.utilities.typing import LambdaContext
+from aws_lambda_powertools import Logger
+
+MODEL_ID = os.environ['MODEL_ID']
+AWS_REGION = os.environ['AWS_REGION']
+LANGUAGE_CODE = os.environ['LANGUAGE_CODE']
+META_TOPICS_STR = os.environ['LABELS']
+META_SENTIMENTS_STR = os.environ['SENTIMENT_LABELS']
+
+INFORMATION_EXTRACTION_PROMPT_SELECTOR = get_information_extraction_prompt_selector(LANGUAGE_CODE)
+TOPIC_MATCH_PROMPT_SELECTOR = get_topic_match_prompt_selector(LANGUAGE_CODE)
+
+TOPIC_MATCH_MODEL_PARAMETERS = {
+    "max_tokens": 500,
+    "temperature": 0.1,
+    "top_k": 20,
+}
+
+INFORMATION_EXTRACTION_MODEL_PARAMETERS = {
+    "max_tokens": 1500,
+    "temperature": 0.1,
+    "top_k": 20,
+}
+
+bedrock_runtime = boto3.client(
+    service_name="bedrock-runtime",
+    region_name=AWS_REGION
+)
+
+logger = Logger()
+
+langchain_core.globals.set_debug(True)
+
+def unknown_to_empty_text_values(extracted_info: ExtractedInformation):
+
+    extracted_info.topic = extracted_info.topic.strip()
+    extracted_info.location = extracted_info.location.strip()
+    extracted_info.sentiment = extracted_info.sentiment.strip()
+
+    extracted_info.topic = extracted_info.topic if extracted_info.topic != "<UNKNOWN>" else ""
+    extracted_info.location = extracted_info.location if extracted_info.location != "<UNKNOWN>" else ""
+    extracted_info.sentiment = extracted_info.sentiment if extracted_info.sentiment != "<UNKNOWN>" else ""
+
+
+def text_topic_match(
+        meta_topics: str,
+        text: str,
+) -> TopicMatch:
+
+    bedrock_llm = ChatBedrock(
+        model_id=MODEL_ID,
+        model_kwargs=TOPIC_MATCH_MODEL_PARAMETERS,
+        client=bedrock_runtime,
+    )
+
+    claude_topic_match_prompt_template = TOPIC_MATCH_PROMPT_SELECTOR.get_prompt(MODEL_ID)
+
+    print(claude_topic_match_prompt_template.format(meta_topics=meta_topics, text=text))
+
+    structured_llm = bedrock_llm.with_structured_output(TopicMatch)
+
+    structured_topic_match_chain = claude_topic_match_prompt_template | structured_llm
+
+    topic_match_obj = structured_topic_match_chain.invoke({"meta_topics": meta_topics, "text": text})
+
+    print("Topic match object")
+    print(type(topic_match_obj))
+    print(topic_match_obj)
+
+    return topic_match_obj
+
+def text_information_extraction(
+        sentiments: str,
+        text: str
+) -> ExtractedInformation:
+
+    bedrock_llm = ChatBedrock(
+        model_id=MODEL_ID,
+        model_kwargs=INFORMATION_EXTRACTION_MODEL_PARAMETERS,
+        client=bedrock_runtime,
+    )
+
+    claude_information_extraction_prompt_template = INFORMATION_EXTRACTION_PROMPT_SELECTOR.get_prompt(MODEL_ID)
+
+    print("The prompt template")
```

Comment:
Is there a reason for using prints when you have a logger instance? It might also be an opportunity to have some logs as INFO and others as DEBUG.

Reply:
Fixed in [e91f7a0](https://github.com/aws-samples/ai-powered-text-insights/pull/3/commits/e91f7a05ccb41059a49562d086e6824bc3af9ecc)